File: BOJ(Java 공부용)/1769.java
import java.io.*;
public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static int[] gox = {0, 1, -1, 0, 1, 1, -1, -1};
    static int[] goy = {1, 0, 0, -1, 1, -1, 1, -1};

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String s = br.readLine();
        int cnt = 0;
        while (s.length() != 1) {
            int sum = 0;
            for (int i = 0; i < s.length(); i++) {
                sum += s.charAt(i) - '0';
            }
            s = String.valueOf(sum);
            cnt++;
        }
        if (Integer.parseInt(s) % 3 == 0) System.out.println(cnt + "\nYES");
        else System.out.println(cnt + "\nNO");
    }
}

File: BOJ(Python 공부용)/1769.py
import sys

input = sys.stdin
sys.setrecursionlimit(100)

n = input.readline().strip("\n")
ok = False
cnt = 0
while len(n) >= 2:
    sum = 0
    p = n
    for i in n:
        sum += int(i)
    cnt += 1
    n = str(sum)

if int(n) % 3 == 0:
    print(cnt, "\nYES")
else:
    print(cnt, "\nNO")